internal/core/subvolume.cpp

```diff
@@ -80,6 +80,24 @@ SubVolume::SubVolume(MetadataHandle const& metadata) {
     this->bounds.upper[sample.dimension()] = sample.nsamples();
 }
 
+
+void SubVolume::constrain(
+    MetadataHandle const& metadata,
```

Comment:
Agree, but keeping a reference on the object itself is also iffy. It's fine in our case as Metadata will always live as long as the SubVolume, but as a general rule this is bad as it can cause dangling references. I opt for keeping the current version